recipes/protobuf/all/conanfile.py

```diff
@@ -45,32 +46,58 @@ def configure(self):
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
-        self._cmake.definitions["protobuf_BUILD_TESTS"] = False
-        self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
-        self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = not self.options.lite
-        self._cmake.definitions["protobuf_BUILD_PROTOBUF_LITE"] = self.options.lite
-        if self.settings.compiler == "Visual Studio":
-            self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in self.settings.compiler.runtime
-        self._cmake.configure(build_folder=self._build_subfolder)
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
+            self._cmake.definitions["protobuf_BUILD_TESTS"] = False
+            self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.get_safe("with_zlib", False)
+            if self.settings.compiler == "Visual Studio":
+                self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in str(self.settings.compiler.runtime)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
-        for patch in self.conan_data["patches"][self.version]:
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            "@_protobuf_FIND_ZLIB@",
+            "# CONAN PATCH _protobuf_FIND_ZLIB@"
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            """COMMAND  protobuf::protoc
+      ARGS --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
+      DEPENDS ${_abs_file} protobuf::protoc""",
+            """COMMAND "${CMAKE_COMMAND}"  #FIXME: use conan binary component
+      ARGS -E env "DYLD_LIBRARY_PATH=${Protobuf_LIB_DIRS}:${CONAN_LIB_DIRS}:${Protobuf_LIB_DIRS_RELEASE}:${Protobuf_LIB_DIRS_DEBUG}:${Protobuf_LIB_DIRS_RELWITHDEBINFO}:${Protobuf_LIB_DIRS_MINSIZEREL}" protoc --${protobuf_generate_LANGUAGE}_out ${_dll_export_decl}${protobuf_generate_PROTOC_OUT_DIR} ${_protobuf_include_path} ${_abs_file}
+      DEPENDS ${_abs_file} USES_TERMINAL"""
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-module.cmake.in"),
+            '# Environment',
+            'if(0)',
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-module.cmake.in"),
+            '# Version info variable',
+            'endif()',
+        )
```

Comment:
I think the call to `find_package_handle_standard_args(Protobuf ...` should also be removed. Conan FindProtobuf.cmake should be in charge of this. 
```suggestion
        )
        )
        tools.replace_in_file(
            os.path.join(self._source_subfolder, "cmake", "protobuf-module.cmake.in"),
            """include(FindPackageHandleStandardArgs)
FIND_PACKAGE_HANDLE_STANDARD_ARGS(Protobuf
    REQUIRED_VARS Protobuf_PROTOC_EXECUTABLE Protobuf_LIBRARIES Protobuf_INCLUDE_DIRS
    VERSION_VAR Protobuf_VERSION
)""",
            """# CONAN PATCH include(FindPackageHandleStandardArgs)
# CONAN PATCH FIND_PACKAGE_HANDLE_STANDARD_ARGS(Protobuf
# CONAN PATCH     REQUIRED_VARS Protobuf_PROTOC_EXECUTABLE Protobuf_LIBRARIES Protobuf_INCLUDE_DIRS
# CONAN PATCH     VERSION_VAR Protobuf_VERSION
# CONAN PATCH )""",
```

Reply:
@Croydon 
Has this problem been resolved?

Reply:
I haven't disabled these calls. I have just added a reference to this review to the issue https://github.com/conan-io/conan-center-index/issues/3492

Reply:
If I’m not wrong, this problem was addressed here https://github.com/conan-io/conan-center-index/pull/2037/commits/4884c45da0fb235c78a641f457023aed5c5f9d21#diff-5dfb647bf8fb82769ddb94143d6b47475c5d4d2db2346115ee6b1991a6db6445R115 by moving the endif() after the find_package_standard_args.